api/src/OpenTelemetry/Internal/Logs/Types.hs

```diff
@@ -111,7 +111,7 @@ data LogRecordProcessor body = LogRecordProcessor
   -- ^ Called when a LogRecord is emitted. This method is called synchronously on the thread that emitted the LogRecord, therefore it SHOULD NOT block or throw exceptions.
   --
   -- A LogRecordProcessor may freely modify logRecord for the duration of the OnEmit call. If logRecord is needed after OnEmit returns (i.e. for asynchronous processing) only reads are permitted.
-  , logRecordProcessorShutdown :: IO (Async ShutdownResult)
+  , logRecordProcessorShutdown :: IO ShutdownResult
```

Comment:
It makes more sense to me for the `LoggerProvider` or a `Processor` to internally implement or choose not to implement concurrency than for it to be enforced by the types. I imagine there are situations where one would not want something to be asynchronous. Removing `Async` from the signature also allows combinators like `withAsync` and `mapConcurrently` to be used.

Reply:
At least for tracing, the docs say this:

> Shutdown can be implemented as a blocking API or an asynchronous API which notifies the caller via a callback or an event. OpenTelemetry client authors can decide if they want to make the shutdown timeout configurable.

I can appreciate the argument in the other direction, but would prefer to keep `Async` here to stay consistent with the SpanProcessor API https://hackage.haskell.org/package/hs-opentelemetry-api-0.1.0.0/docs/OpenTelemetry-Processor.html#t:Processor

Reply:
Another reason I made the change is that having `Async` in the signature made it much more difficult to propagate and combine `ExportResult`s, `FlushResult`s, and `ShutdownResult`s. I also figured that it would be good if either `shutdown`, `forceFlush`, and `export` were either all asynchronous or all synchronous.

Also, the way I rewrote `shutdownLoggerProvider` seems cleaner and more abstract (using `forConcurrently`) than the way it was written for `shutdownTracerProvider` before. Plus, people defining a `LogRecordProcessor` no longer have to worry about the asynchronous part because it is handled by the `LoggerProvider`.

Let me know your thoughts.